axelor-sale/src/main/java/com/axelor/apps/sale/service/saleorder/SaleOrderLineGeneratorServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+  protected SaleOrderLineProductService saleOrderLineProductService;
+  protected SaleOrderLineRepository saleOrderLineRepository;
+  protected ProductRepository productRepository;
+  protected SaleOrderRepository saleOrderRepository;
+  protected SaleOrderLineComputeService saleOrderLineComputeService;
+  protected SaleOrderComputeService saleOrderComputeService;
+  protected SaleOrderLineDomainService saleOrderLineDomainService;
+  protected SaleOrderService saleOrderService;
+  protected SaleOrderOnLineChangeService saleOrderOnLineChangeService;
+  protected AppBaseService appBaseService;
+
+  @Inject
+  public SaleOrderLineGeneratorServiceImpl(
+      SaleOrderLineInitValueService saleOrderLineInitValueService,
+      SaleOrderLineProductService saleOrderLineProductService,
+      SaleOrderRepository saleOrderRepository,
+      SaleOrderLineRepository saleOrderLineRepository,
+      SaleOrderLineComputeService saleOrderLineComputeService,
+      SaleOrderComputeService saleOrderComputeService,
+      SaleOrderLineDomainService saleOrderLineDomainService,
+      ProductRepository productRepository,
+      SaleOrderService saleOrderService,
+      AppBaseService appBaseService,
+      SaleOrderOnLineChangeService saleOrderOnLineChangeService) {
+    this.saleOrderLineInitValueService = saleOrderLineInitValueService;
+    this.saleOrderLineProductService = saleOrderLineProductService;
+    this.saleOrderLineRepository = saleOrderLineRepository;
+    this.saleOrderRepository = saleOrderRepository;
+    this.saleOrderLineComputeService = saleOrderLineComputeService;
+    this.saleOrderComputeService = saleOrderComputeService;
+    this.saleOrderLineDomainService = saleOrderLineDomainService;
+    this.productRepository = productRepository;
+    this.saleOrderService = saleOrderService;
+    this.saleOrderOnLineChangeService = saleOrderOnLineChangeService;
+    this.appBaseService = appBaseService;
+  }
+
+  @Transactional(rollbackOn = {Exception.class})
+  @Override
+  public SaleOrderLine createSaleOrderLine(SaleOrder saleOrder, Product product)
+      throws AxelorException {
+    chackSaleOrderAndProduct(saleOrder, product);
+    SaleOrderLine saleOrderLine = new SaleOrderLine();
+    saleOrderLineInitValueService.onNewInitValues(saleOrder, saleOrderLine);
+    checkProduct(saleOrder, saleOrderLine, product);
+    saleOrderLine.setProduct(product);
+    saleOrderLineProductService.computeProductInformation(saleOrderLine, saleOrder);
+    saleOrderLineComputeService.computeValues(saleOrder, saleOrderLine);
+    saleOrderLineRepository.save(saleOrderLine);
+    saleOrder.addSaleOrderLineListItem(saleOrderLine);
+    saleOrderComputeService.computeSaleOrder(saleOrder);
+    saleOrderOnLineChangeService.handleComplementaryProducts(saleOrder);
+    saleOrderRepository.save(saleOrder);
+
+    return saleOrderLine;
+  }
+
+  private static void chackSaleOrderAndProduct(SaleOrder saleOrder, Product product)
```

Comment:
Make the method protected and non static.
Also fix the typo to "check" and not "chack"